src/Columns/ColumnArray.h

```diff
@@ -141,6 +142,16 @@ class ColumnArray final : public COWHelper<IColumn, ColumnArray>
         return false;
     }
 
+    double getRatioOfDefaultRows(double sample_ratio) const override
+    {
+        return getRatioOfDefaultRowsImpl<ColumnArray>(sample_ratio);
+    }
+
+    void getIndicesOfNonDefaultRows(Offsets & indices, size_t from, size_t limit) const override
+    {
+        return getIndicesOfNonDefaultRowsImpl<ColumnArray>(indices, from, limit);
+    }
+
```

Comment:
I would better move implementation to cpp. Probably it is not good to instantiate templates here